osu.Game.Rulesets.Osu/Difficulty/Preprocessing/OsuDifficultyHitObject.cs

```diff
@@ -74,10 +75,10 @@ public class OsuDifficultyHitObject : DifficultyHitObject
         private readonly OsuHitObject lastLastObject;
         private readonly OsuHitObject lastObject;
 
-        public OsuDifficultyHitObject(HitObject hitObject, HitObject lastLastObject, HitObject lastObject, double clockRate)
-            : base(hitObject, lastObject, clockRate)
+        public OsuDifficultyHitObject(HitObject hitObject, HitObject lastObject, double clockRate, List<DifficultyHitObject> objects, int position)
+            : base(hitObject, lastObject, clockRate, objects, position)
         {
-            this.lastLastObject = (OsuHitObject)lastLastObject;
+            lastLastObject = position > 1 ? (OsuHitObject)Previous(1).BaseObject : null;
```

Comment:
Could simplify this to `lastLastObject = (OsuHitObject)Previous(1)?.BaseObject`, if `Previous` allows returning null when index is out of range (see other comment).